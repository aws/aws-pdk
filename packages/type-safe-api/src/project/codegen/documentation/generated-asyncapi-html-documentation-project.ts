/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { Project, ProjectOptions, Task } from "projen";
import { GeneratedAsyncApiHtmlDocumentationOptions } from "../../types";
import { TypeSafeApiCommandEnvironment } from "../components/type-safe-api-command-environment";

export interface GeneratedAsyncApiHtmlDocumentationProjectOptions
  extends ProjectOptions,
    GeneratedAsyncApiHtmlDocumentationOptions {
  /**
   * Path to the AsyncAPI Specification for which to generate docs, relative to the project outdir
   */
  readonly specPath: string;
}

export class GeneratedAsyncApiHtmlDocumentationProject extends Project {
  private readonly generateTask: Task;

  constructor(options: GeneratedAsyncApiHtmlDocumentationProjectOptions) {
    super(options);
    TypeSafeApiCommandEnvironment.ensure(this);

    this.generateTask = this.addTask("generate");
    this.generateTask.exec(
      `npx --yes @asyncapi/cli@1.7.3 generate fromTemplate "${options.specPath}" @asyncapi/html-template@2.3.2 --param singleFile=true --param outFilename=index.html --force-write`
    );

    this.compileTask.spawn(this.generateTask);

    if (!options.commitGeneratedCode) {
      this.gitignore.addPatterns("index.html");
    }
  }
}
