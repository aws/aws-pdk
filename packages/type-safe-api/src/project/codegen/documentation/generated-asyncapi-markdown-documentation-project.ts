/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { Project, ProjectOptions, Task } from "projen";
import { GeneratedAsyncApiMarkdownDocumentationOptions } from "../../types";
import { TypeSafeApiCommandEnvironment } from "../components/type-safe-api-command-environment";

export interface GeneratedAsyncApiMarkdownDocumentationProjectOptions
  extends ProjectOptions,
    GeneratedAsyncApiMarkdownDocumentationOptions {
  /**
   * Path to the AsyncAPI Specification for which to generate docs, relative to the project outdir
   */
  readonly specPath: string;
}

export class GeneratedAsyncApiMarkdownDocumentationProject extends Project {
  private readonly generateTask: Task;

  constructor(options: GeneratedAsyncApiMarkdownDocumentationProjectOptions) {
    super(options);
    TypeSafeApiCommandEnvironment.ensure(this);

    this.generateTask = this.addTask("generate");
    this.generateTask.exec(
      `npx --yes @asyncapi/cli@1.7.3 generate fromTemplate "${options.specPath}" @asyncapi/markdown-template@1.5.0 --param outFilename=index.md --force-write`
    );
    this.compileTask.spawn(this.generateTask);

    if (!options.commitGeneratedCode) {
      this.gitignore.addPatterns("index.md");
    }
  }
}
