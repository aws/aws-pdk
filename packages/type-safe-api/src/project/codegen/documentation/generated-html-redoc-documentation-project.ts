/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import * as path from "path";
import { Project, ProjectOptions, Task } from "projen";
import { buildInvokeCustomDocsGeneratorCommand } from "../components/utils";

export interface GeneratedHtmlRedocDocumentationProjectOptions
  extends ProjectOptions {
  /**
   * Path to the OpenAPI Specification for which to generate docs, relative to the project outdir
   */
  readonly specPath: string;
}

export class GeneratedHtmlRedocDocumentationProject extends Project {
  private readonly generateTask: Task;

  constructor(options: GeneratedHtmlRedocDocumentationProjectOptions) {
    super(options);

    this.generateTask = this.addTask("generate");

    const cmd = buildInvokeCustomDocsGeneratorCommand({
      generator: "html-redoc",
      specPath: options.specPath,
      outputPath: this.outdir,
    });
    this.generateTask.exec(cmd.command, {
      cwd: path.relative(this.outdir, cmd.workingDir),
    });

    this.compileTask.spawn(this.generateTask);

    this.gitignore.addPatterns("index.html");
  }
}
