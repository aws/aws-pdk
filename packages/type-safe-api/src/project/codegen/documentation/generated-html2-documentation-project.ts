/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { Project, ProjectOptions, Task } from "projen";
import { DocumentationFormat } from "../../languages";
import { GeneratedHtml2DocumentationOptions } from "../../types";
import { OpenApiToolsJsonFile } from "../components/open-api-tools-json-file";
import {
  buildInvokeOpenApiGeneratorCommandArgs,
  buildTypeSafeApiExecCommand,
  OtherGenerators,
  TypeSafeApiScript,
} from "../components/utils";

export interface GeneratedHtml2DocumentationProjectOptions
  extends ProjectOptions,
    GeneratedHtml2DocumentationOptions {
  /**
   * Path to the OpenAPI Specification for which to generate docs, relative to the project outdir
   */
  readonly specPath: string;
}

export class GeneratedHtml2DocumentationProject extends Project {
  private readonly generateTask: Task;

  constructor(options: GeneratedHtml2DocumentationProjectOptions) {
    super(options);

    // Add OpenAPI Generator cli configuration
    OpenApiToolsJsonFile.ensure(this).addOpenApiGeneratorCliConfig(
      options.openApiGeneratorCliConfig
    );

    this.generateTask = this.addTask("generate");
    this.generateTask.exec(
      buildTypeSafeApiExecCommand(
        TypeSafeApiScript.GENERATE,
        buildInvokeOpenApiGeneratorCommandArgs({
          generator: DocumentationFormat.HTML2,
          specPath: options.specPath,
          generatorDirectory: OtherGenerators.DOCS,
        })
      )
    );

    this.compileTask.spawn(this.generateTask);

    this.gitignore.addPatterns(".openapi-generator", "index.html");
  }
}
