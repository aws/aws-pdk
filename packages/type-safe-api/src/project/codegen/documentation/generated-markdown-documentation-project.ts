/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { Project, ProjectOptions, Task } from "projen";
import { GeneratedMarkdownDocumentationOptions } from "../../types";
import { TypeSafeApiCommandEnvironment } from "../components/type-safe-api-command-environment";
import {
  buildCodegenCommandArgs,
  buildTypeSafeApiExecCommand,
  TypeSafeApiScript,
} from "../components/utils";

export interface GeneratedMarkdownDocumentationProjectOptions
  extends ProjectOptions,
    GeneratedMarkdownDocumentationOptions {
  /**
   * Path to the OpenAPI Specification for which to generate docs, relative to the project outdir
   */
  readonly specPath: string;
}

export class GeneratedMarkdownDocumentationProject extends Project {
  private readonly generateTask: Task;

  constructor(options: GeneratedMarkdownDocumentationProjectOptions) {
    super(options);
    TypeSafeApiCommandEnvironment.ensure(this);

    this.generateTask = this.addTask("generate");
    this.generateTask.exec(
      buildTypeSafeApiExecCommand(
        TypeSafeApiScript.GENERATE,
        buildCodegenCommandArgs({
          specPath: options.specPath,
          templateDirs: ["docs/templates/markdown"],
        })
      )
    );

    this.compileTask.spawn(this.generateTask);

    if (!options.commitGeneratedCode) {
      this.gitignore.addPatterns("Apis", "Models", "README.md");
    }

    this.gitignore.addPatterns(".openapi-generator", ".tsapi-manifest");
  }
}
