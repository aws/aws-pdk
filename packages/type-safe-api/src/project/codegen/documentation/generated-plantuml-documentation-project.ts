/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import * as path from "path";
import { Project, ProjectOptions, Task } from "projen";
import { DocumentationFormat } from "../../languages";
import {
  buildInvokeOpenApiGeneratorCommand,
  NonClientGeneratorDirectory,
} from "../components/utils";

export interface GeneratedPlantumlDocumentationProjectOptions
  extends ProjectOptions {
  /**
   * Path to the OpenAPI Specification for which to generate docs, relative to the project outdir
   */
  readonly specPath: string;
}

export class GeneratedPlantumlDocumentationProject extends Project {
  private readonly generateTask: Task;

  constructor(options: GeneratedPlantumlDocumentationProjectOptions) {
    super(options);

    this.generateTask = this.addTask("generate");

    const cmd = buildInvokeOpenApiGeneratorCommand({
      generator: DocumentationFormat.PLANTUML,
      specPath: options.specPath,
      outputPath: this.outdir,
      generatorDirectory: NonClientGeneratorDirectory.DOCS,
    });
    this.generateTask.exec(cmd.command, {
      cwd: path.relative(this.outdir, cmd.workingDir),
    });

    this.compileTask.spawn(this.generateTask);

    this.gitignore.addPatterns(".openapi-generator", "schemas.plantuml");
  }
}
