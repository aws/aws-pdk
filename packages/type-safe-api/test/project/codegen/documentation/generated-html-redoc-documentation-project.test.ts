/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import path from "path";
import { GeneratedHtmlRedocDocumentationProject } from "../../../../src/project/codegen/documentation/generated-html-redoc-documentation-project";
import { synthProject } from "../../snapshot-utils";

describe("GeneratedHtmlRedocDocumentationProject Unit Tests", () => {
  it("Synth", () => {
    const project = new GeneratedHtmlRedocDocumentationProject({
      outdir: path.resolve(__dirname, "html-redoc-docs"),
      name: "docs",
      specPath: "my-spec.json",
    });
    expect(synthProject(project)).toMatchSnapshot();
  });
});
