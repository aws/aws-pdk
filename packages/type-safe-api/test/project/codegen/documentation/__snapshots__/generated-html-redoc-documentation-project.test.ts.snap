// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneratedHtmlRedocDocumentationProject Unit Tests Synth 1`] = `
Object {
  ".gitignore": "# ~~ Generated by projen. To modify, edit .projenrc.js and run \\"npx projen\\".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
index.html
",
  ".projen/files.json": Object {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \\"npx projen\\".",
    "files": Array [
      ".gitattributes",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
    ],
  },
  ".projen/tasks.json": Object {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \\"npx projen\\".",
    "tasks": Object {
      "build": Object {
        "description": "Full release build",
        "name": "build",
        "steps": Array [
          Object {
            "spawn": "default",
          },
          Object {
            "spawn": "pre-compile",
          },
          Object {
            "spawn": "compile",
          },
          Object {
            "spawn": "post-compile",
          },
          Object {
            "spawn": "test",
          },
          Object {
            "spawn": "package",
          },
        ],
      },
      "compile": Object {
        "description": "Only compile",
        "name": "compile",
        "steps": Array [
          Object {
            "spawn": "generate",
          },
        ],
      },
      "default": Object {
        "description": "Synthesize project files",
        "name": "default",
      },
      "eject": Object {
        "description": "Remove projen from the project",
        "env": Object {
          "PROJEN_EJECTING": "true",
        },
        "name": "eject",
        "steps": Array [
          Object {
            "spawn": "default",
          },
        ],
      },
      "generate": Object {
        "name": "generate",
        "steps": Array [
          Object {
            "cwd": "../../../../../scripts/custom/docs",
            "exec": "./html-redoc --spec-path ../../../test/project/codegen/documentation/html-redoc-docs/my-spec.json --output-path ../../../test/project/codegen/documentation/html-redoc-docs",
          },
        ],
      },
      "package": Object {
        "description": "Creates the distribution package",
        "name": "package",
      },
      "post-compile": Object {
        "description": "Runs after successful compilation",
        "name": "post-compile",
      },
      "pre-compile": Object {
        "description": "Prepare the project for compilation",
        "name": "pre-compile",
      },
      "test": Object {
        "description": "Run tests",
        "name": "test",
      },
    },
  },
}
`;
